src/software/ai/intent/intent.h

```diff
@@ -93,6 +110,7 @@ class Intent
     virtual ~Intent() = default;
 
    private:
+    unsigned int robot_id;
```

Comment:
Quick comment like `The id of the robot this intent is to be run on` would be helpful here.

Reply:
added